src/jvm/main/org/jetbrains/kotlinx/lincheck/paramgen/ParameterGenerators.kt

```diff
@@ -0,0 +1,246 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.paramgen
+
+import org.jetbrains.kotlinx.lincheck.RandomProvider
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import java.util.*
+
+/**
+ * The implementation of this interface is used to generate parameters
+ * for [operation][Operation].
+ *
+ * Now all successors must have constructor with [RandomProvider] and config [String].
+ */
+interface ParameterGenerator<T> {
+    fun generate(): T
+
+    /**
+     * Resets current range bounds to start, if this generator has such expanding range.
+     * Meanwhile, it shouldn't reset random to avoid undesired correlation between scenarios.
+     */
+    fun resetRange()
+    class Dummy : ParameterGenerator<Any?> {
+        override fun generate(): Any {
+            throw UnsupportedOperationException()
+        }
+
+        override fun resetRange() {}
+    }
+}
+
+class IntGen(randomProvider: RandomProvider, configuration: String) : ParameterGenerator<Int> {
+    private val generator: ExpandingRangeIntGenerator
+
+    init {
+        generator = ExpandingRangeIntGenerator(
+            random = randomProvider.createRandom(),
+            configuration = configuration,
+            minStartInclusive = Int.MIN_VALUE,
+            maxEndInclusive = Int.MAX_VALUE,
+            type = "int"
+        )
+    }
+
+    override fun generate(): Int = generator.nextInt()
+    override fun resetRange() = generator.reset()
+}
+
+class BooleanGen(randomProvider: RandomProvider, configuration: String) : ParameterGenerator<Boolean> {
+    private val random = randomProvider.createRandom()
+
+    override fun generate() = random.nextBoolean()
+    override fun resetRange() {
+    }
+}
+
+class ByteGen(randomProvider: RandomProvider, configuration: String) : ParameterGenerator<Byte> {
+    private val generator: ExpandingRangeIntGenerator
+
+    init {
+        generator = ExpandingRangeIntGenerator(
+            random = randomProvider.createRandom(),
+            configuration = configuration,
```

Comment:
You specify the bounds below but the configuration may have wider bounds. Do you check this? If so, please add a comment indicating that.